### PR TITLE
[WIP] feat(http): adding a skeleton for a new mock http source helper

### LINKED ISSUE #567: Add mockHTTPSource helper
It would be nice if the HTTP driver provided a helper for mocking the HTTP source, comparable to mockDOMSource or mockTimeSource.

The API could be comparable to mockDOMSource.

Maybe:

```js
const response$ = xs.of({data: {some: 'thing'}});

const HTTP = mockHTTPSource({
  'categoryName': xs.of(response$)
});
```

### PR DESCRIPTION
**Pull request by Widdershin**
Carrying on from #571, had to make a new pull request because I couldn't push to the old branch.

Fixes #567

### CODE REVIEW

File: http/test/node/mock-http-source.ts [hunk 1]
```diff
@@ -1,63 +1,23 @@
...
-      complete: () => done(),
-    });
-  });
-
-  it('supports categorized streams', (done) => {
-    const response: any = {
-      text: 'Hello world',
+import {setup} from '../../../run/src';
+
+describe.only('mockHTTPDriver', function() {
```
Comment (staltz) [inline http/test/node/mock-http-source.ts]: I suppose `describe.only` is temporary? :)

Comment (Widdershin) [inline http/test/node/mock-http-source.ts]: Most definitely. This latest commit doesn't even compile yet 😸 

I need a way to turn a `Partial<Response>` into something that implement the `Response` interface. Ideally I want to match superagent's behaviour perfectly, but most of the logic I need to utilize is tightly coupled with performing side effects.

In the past I've used https://github.com/M6Web/superagent-mock, so hopefully I can reuse some of the work they've done.

**Comment by grushetsky** [comment]
@Widdershin, do you need any help on this one? We'd like to use mocked HTTP source too!

**Comment by Widdershin** [comment]
@grushetsky I'd love some help, been meaning to finish this off for a while but I've been a bit swamped.

The next main thing that needs to be done is finishing the implementation of `partialResponseToMockResponse` (https://github.com/cyclejs/cyclejs/pull/650/files#diff-f0a89f91a42b58374b1c8cec24cd397cR59). This effectively needs to replicate the behaviour of the superagent internals. I was going to looking into extracting part of [superagent-mock](https://github.com/M6Web/superagent-mock), but I have yet to do that.

Hasty explanation so let me know if you want more detail, or if that's enough to go on.

**Comment by jvanbruegge** [comment]
Made irrelevant due to the rewrite

**Pull request closed without merging**